### ISSUE TITLE
[Feature Request]RetroAchievements

### ISSUE THREAD

**Issue by JakobDev** [Arisotura/melonDS#1146]
[RetroAchievements](https://retroachievements.org) is a site which add Achievements for retro games (like the name says). You can get the Achievements if you play the Game in a Emulator that supports RetroAchievements. Would be nice if you could add support for RetroAchievements in melonDS.

**Comment by Arisotura** [comment]
what's the point

**Comment by JakobDev** [comment]
If you use RetroAchievements you get Achievements in Games. For example, you take the first Sword in The Legend of Zelda.
![grafik](https://user-images.githubusercontent.com/15185051/124459771-d004c480-dd8e-11eb-9fc8-8fc69e3fffef.png)
This requires of course support of the Emulator. RetroAchievements already offers Achievements for DS Games e.g [New Super Mario Bros](https://retroachievements.org/game/11729). What I request is adding support for this Achievements in melonDS. For example, I would get the Achievement Grassy Fields when I complete World 1 in New Super Mario Bros in melonDS.

**Comment by Arisotura** [comment]
I mean, I don't get the point of achievements to begin with, and even less so modifying an emulator to hack into the game to detect these things, which sounds like asking for trouble for little gain from a technical standpoint.

I just want to see what the other team members think of this.

**Comment by Arisotura** [comment]
after discussion, it appears that Retroarch's implementation of that feature is shitty and I don't want to deal with issue reports stemming from that.

that being said, we here only support the standalone desktop version of melonDS. the Retroarch port is maintained by someone else, and it seems like supporting retroachievements would be their work.

**Comment by MojoJojoDojo** [comment]
You can use retroachievements in standalone emulators though.
Duckstation got an implementation that does not use any libretro stuff or api (works on standalone).

**Comment by FalcusDev** [comment]
To further add on to this, Retroachievements isn't specifically tied to RetroArch (it is completely separate from RetroArch, it just happened to be implemented there about 6 or 7 years after the project started) however having achievement support on MelonDS standalone would be nice for a few reasons, the main ones being the extra functionality that has been integrated into MelonDS like microphone and DSi support.
On top of this it would also give those who would like to earn achievements some additional options in which emulators they use and not have to rely on RetroArch for as many platforms. At the moment there has been a bit more standalone support that is coming like PCSX2 as well as a build having been made for WinArcadia, on top of some which have been around for a while like Meka, Project64, VBA etc.

I can understand if you'd rather focus on other things, have no interest in this etc however and will appreciate the work you have all put into this regardless.

Thank you regardless of what comes from here.

**Comment by XParadisiac** [comment]
+1 for Retro achievement support! RetroArch does not function with S-Pen, but MelonDS does. Would be fantastic to earn achievements with it!

**Comment by CoreTaku** [comment]
+2 for RA support. My Retroid Pocket 3+ needs a dependable emulator for me to earn my spicy, spicy achievement challenges with.

**Comment by FoxbitDreamtail** [comment]
+3 for RA Support, it gives me a sense of accomplishment and gives me that slight dopamine push in a grueling/grindy game such as NSMBDS

**Comment by Bastigonzales** [comment]
+4

**Comment by AftDawn** [comment]
+5 because the retroarch core version of melon doesn't expose wifi, but has RA, and i wanna get the full experience of heartgold again

**Comment by nighto** [comment]
Reviving this thread to mention that the [melonDS Android fork is implementing RetroAchievements support](https://github.com/rafaelvcaetano/melonDS-android/discussions/1117). Hope you could at some point revisit your opinion about that matter and that it ever gets backported to the main melonDS. :)

**Comment by FoxbitDreamtail** [comment]
> Reviving this thread to mention that the [melonDS Android fork is implementing RetroAchievements support](https://github.com/rafaelvcaetano/melonDS-android/discussions/1117). Hope you could at some point revisit your opinion about that matter and that it ever gets backported to the main melonDS. :)

This seems all fine and dandy until you realize that this would be difficult to implement properly on desktop, Android source code and x86 source code looks vastly different, primarily in implementing features, if you have source code that can compile both it must be a hybrid that can or have certain configuration files for flags to have a compile that way. I'm not trying to burst your bubble or nothing it's just, I'm not getting my hopes up for it. Not to mention it performs terribly on a core 2 4400 paired with a NVIDIA GTX 550 TI in Linux anyway.

**Comment by ludg1e** [comment]
As far as I know it could be viable using the rcheevos library from RetroAchievements. Duckstation and PCSX2 are already using it.
https://github.com/RetroAchievements/rcheevos

**Comment by CasualPokePlayer** [comment]
rcheevos just does the platform independent work. It does not handle any kind of UI work and does not handle making the actual http requests, stuff which you cannot just grab from the android fork.

**Comment by ludg1e** [comment]
Yes, I know that

**Comment by AleKing31** [comment]
+6

**Comment by Wesker12345** [comment]
+7

**Comment by retropieuser** [comment]
Hi there, I'd also love to see this feature implemented. It was added into Duckstation and recently PPSSPP. It seems to be getting added into Dolphin as well. It would be great to see on MelonDS as the options for DS emulators is quite limited and frankly the further we can get away from Retroarch and that environment is a positive, as it just seems so bogged down

**Comment by SentaiBrad** [comment]
There is a massive point. They're fun first and foremost and add something cool for new-to-you games or old favorites. I get the development cost, but they're clearly becoming something people want.

**Comment by RetroFlock** [comment]
Let's Do It! I Love Retroachevmints!

**Comment by Oceanman0123** [comment]
Please add retroachievements to MelonDS

**Comment by TakaOtaku** [comment]
Would also love it!

**Comment by Optekah** [comment]
Any news / reconsideration?

Myself and a ton of others would love to never need to install the god awfulness that is RetroArch ever again. Supporting standalone emulators is definitely the way to go, but seeing so many of them just completely disregard achievements makes me sad.

Thanks for your hard work though devs!

**Comment by mblithium** [comment]
I think the addition of Retroachivements is interesting for MelonDS, but we have to consider that the frontends created for it usually bring this function (like the MelonDS port for Android and Retroarch) and so it's not urgent, the developers of the project need to focus on the core (which is already a complicated task when it comes to emulators). I think someone can do an implementation and send a pull request anyway.

**Comment by PlasmaStormZ** [comment]
Reviving this thread once more. I would absolutely love for Retroacheivements to be on the desktop version of MelonDS, I love that it is on the Android version so I have high hopes that it can be done on the desktop version. I'm all for it! +15!

**Comment by Amnesia1000** [comment]
I join the request, I understand that there are other more important issues that they are working on, but it would be nice to have this functionality in the real melonDS. Without having to resort to retroarch.

**Comment by Venaci** [comment]
I join the request. I have context you may not like, but may want to hear.

It's a little long, but I urge the devs to read. I typed it because I care, not to hear myself "talk". 

The only way DSi is currently is supported for achievements is through BizHawk. You can PLAY DSi on RetroArch, but chevos are not yet supported through it, AFAIK. To incorporate DSi games into the NAND, we are told to use MelonDS to include it, or a standalone app.

I've never used MelonDS, at least standalone. I was brought here because I thought something this popular would have it. RetroArch, RALibRetro, PCSX2, BizHawk, and now even Dolphin has it. I read the whole thread, and MANY people have voiced wanting it added. 

I've done programming. I was one of the first to great a Pokemon Gen2 clock reset calculator 2001 in C++. I didn't publish it because I couldn't create a GUI: I was systems only. I knew my audience wouldn't be good at using the app in the console. That said, I think you underestimate how many players only use emulators with it. That's almost 53K users who have enough score to be ranked that you're excluding from using your app. That's going to increase A LOT once Gamecube achievement sets are out.

I'm not going to whisper "thank you" for something that I don't use, but I know you shared it with the community because you care. I'm a person with compassion who sees you not taking requests seriously, because you yourself aren't interested in it. I think you miss that nobody wants to say is that No$gb is no deprecated, and that Melon DS may end up meeting the same fate if it doesn't include support for achievements on a site with more players than who use your app. 

I'm not saying this to be rude or forceful, just that at some point you have to pick if you're doing it for yourself or others. In this case, it seems it's completely for yourself. If you have been developing this solely for yourselves or passion, you're completely free to do what you will. But I'll be the one to say that devs (at least Arisotura) is stubborn and ignoring what people want. If you want your project to be used more, and people ask for something, you need to consider it without whatever bias you have going into it.

IE, dealing with reports. You can easily just direct them to the RetroAchievement forums, except in cases that it's actually on your side. There's little front-end, and you can program either alt functions for using achievements, or just have a switch that's set to 0 when it's off. This would prevent performance issues, except an increase in filesize or maybe a 1% slowdown.

All I'm saying is that it's a bigger deal than you may think. It's a NIGHTMARE setting up BizHawk, so if you made this work with RetroAchievements, you'd basically have a monopoly on the DSi emulator market how PCSX2 does!

**Comment by JesseTG** [comment]
Psst...[melonDS DS](https://github.com/JesseTG/melonds-ds) supports achievements _and_ easy installation for DSiWare.

**Comment by Riku55** [comment]
Another +1 here. While with melonDS DS, there is a nice alternative available, there are situations where a standalone is just preferable to a libretro core. Please give this another thought for the sake of the community :)

**Comment by Zekromaster** [comment]
If someone actually made a fork incorporating rcheevos support instead of just offloading it onto you, would you be open to the idea of upstreaming the changes?

**Comment by hipoleto** [comment]
I join this, having achievements inside emulators is like an incentive to actually replay most games, i understand it might be difficult but please reconsider about working with the retroachievements support, many people would like that feature and instead of having to use retroarch as its more convenient to most, just using the stand alone emulator.

**Comment by CoreTaku** [comment]
I placed my +2 for RA Support a long time ago. Can I place my +8 now? Is there a reset timer?

[x

Nah, seriously. After reading these over the months, and all things considered, I'd understand if this never gets done, 'cause of it being a pain in the butt, BUT just know that if it were to go down that it would be GREATLY appreciated. Like the appreciation would be spouting harder than a Dr. Pepper fountain. Trust me. The RA community is growing fast.

**Comment by sks316** [comment]
The Dolphin emulator (standalone emulator for Wii and GameCube) just pushed RetroAchievements support, so I was hoping the melonDS dev team would consider revisiting this, as an implementation in a big emulator like Dolphin shows real interest in RetroAchievements.
To counter the "what's the point" comment, the point is to have fun.

**Comment by Optekah** [comment]
> an implementation in a big emulator like Dolphin shows real interest

Definitely lots of interest. We currently have achievements for 52 (!) different consoles. There was such a hype around the newest gamecube achievements that the site literally crashed due to the traffic, which is insane.

**Comment by JesseTG** [comment]
> @JesseTG You could open a PR to upstream your changes

Most of the work is in RetroArch itself.

**Comment by nighto** [comment]
Not really: at this moment only Dolphin stand-alone is suported; the
Dolphin core in RetroArch is _not_ supported.

Em ter., 16 de jul. de 2024 às 19:24, Jesse Talavera <
***@***.***> escreveu:

**Comment by Luna-Karch** [comment]
Having achievements on melonds would be awesome so I don't have to use retroarch

**Comment by SaraJLeen** [comment]
RetroAchievements have nothing to do with RetroArch, and the SDK implementation for them doesn't have to be nearly as messy as how RetroArch can often be. Please support them! It'd make an already wonderful emulator even greater! Or at least, say you'd be open to accepting a PR if someone else were to implement them!

**Comment by yoshiweegee** [comment]
Chiming in to say that RetroAchievements would be huge for melonDS since it would make it the **only** supported DS emulator to also support local multiplayer netplay (once it's finished), which could easily make games with co-op, like _The Legend of Zelda: Four Swords Anniversary Edition_ and _Kirby Super Star Ultra_, much more interesting for achievement-hunting.

**Comment by IzuSensei** [comment]
I'm also praying for the addition of this feature, which so many of us have been waiting for - it would be a revolution!

**Comment by AgentWho007** [comment]
It's been 3 years, I still hope it gets the Retroachievements support because it's easier for me to play DSI games here

**Comment by Zekromaster** [comment]
Some time ago, I managed to get RetroAchievements working in a local copy of MelonDS, although I didn't develop it further than a basic prototype with code I wouldn't push to a repo meant for long term maintenance at the time. It's basically a prototype.

I got achievements to unlock and a configuration GUI, and the big issues that I had pending are two technical and two functional:

**Technical**
* RetroAchievements support creates a dependency on some kind of http networking library. I guess libCURL is in the environment, but I used CPR for my prototype. It's probably perfectly possible to switch to raw libCURL.
* RetroAchievements should be kept mostly a frontend-side thing, especially since there are other frontends that use it. I couldn't find an elegant way to do this, from what I remember I just slapped an EventBus on it for the sake of making it work. Adding a third codebase would probably be even worse. This also can probably be solved, I just didn't see a good solution at the time.

**Functional**
* I still didn't have a GUI that lists achievements while playing
* Minorly, although hardcore mode is trivial to implement by altering the frontend so that cheats and savestates are unavailable, there's still work to be done on it

There also the fact that my prototype was using the default OSD as a notification system since the frontend doesn't have a proper "good looking" notification system, but seeing as Dolphin did something similar, this can probably be an acceptable choice in a definitive version of the feature.

All in all, this is more doable than it looks and I could look back into finishing it, although I wouldn't bother if there was no interest in getting RetroAchievements into the upstream - the RA community has mostly abandoned the idea of "RetroAchievements forks" for emulators, which is a big reason why I had decided the first step for returning to this should've been getting a "go-ahead" for submitting a PR.

**Comment by HylianEternal** [comment]
PLEASE give this person the go-ahead. I see this as a win-win for everyone.

> Some time ago, I managed to get RetroAchievements working in a local copy of MelonDS, although I didn't develop it further than a basic prototype with code I wouldn't push to a repo meant for long term maintenance at the time. It's basically a prototype.
> 
> I got achievements to unlock and a configuration GUI, and the big issues that I had pending are two technical and two functional:
> 
> **Technical**
> * RetroAchievements support creates a dependency on some kind of http networking library. I guess libCURL is in the environment, but I used CPR for my prototype. It's probably perfectly possible to switch to raw libCURL.
> * RetroAchievements should be kept mostly a frontend-side thing, especially since there are other frontends that use it. I couldn't find an elegant way to do this, from what I remember I just slapped an EventBus on it for the sake of making it work. Adding a third codebase would probably be even worse. This also can probably be solved, I just didn't see a good solution at the time.
> 
> **Functional**
> * I still didn't have a GUI that lists achievements while playing
> * Minorly, although hardcore mode is trivial to implement by altering the frontend so that cheats and savestates are unavailable, there's still work to be done on it
> 
> There also the fact that my prototype was using the default OSD as a notification system since the frontend doesn't have a proper "good looking" notification system, but seeing as Dolphin did something similar, this can probably be an acceptable choice in a definitive version of the feature.
> 
> All in all, this is more doable than it looks and I could look back into finishing it, although I wouldn't bother if there was no interest in getting RetroAchievements into the upstream - the RA community has mostly abandoned the idea of "RetroAchievements forks" for emulators, which is a big reason why I had decided the first step for returning to this should've been getting a "go-ahead" for submitting a PR.

**Comment by Optekah** [comment]
Support for RA has now surpassed all others and become the **single most discussed open issue** on the GitHub.

Would love the green light!

**Comment by Zekromaster** [comment]
(Not to curb anyone's enthusiasm, but be warned that even if I resume my work I'll take my time AND everything would also still conditional on testing and quality control on RetroAchievements' side before it gets into any user-facing release build)

**Comment by ksfaltus** [comment]
2025, with proper dual screen support, the only thing stopping me moving to standalone MelonDS full time is lack of achievement support.

**Comment by SentaiBrad** [comment]
> 2025, with proper dual screen support, the only thing stopping me moving to standalone MelonDS full time is lack of achievement support.

Same. I have a portable monitor second screen setup for my Steam Deck, and my DNA Duo screen has been ordered for almost a year now. The only reason why I use MelonDS DS in RetroArch is RetroAchievements, but it lacks the DS style. Would prefer standalone for screen and RA. I keep checking this thread every few weeks / months xD

**Comment by NitroShellMKDS** [comment]
Please give it the greenlight

**Comment by chowder908** [comment]
The unofficial android port got it working I don't see why we still haven't got it yet. :(

**Comment by SuperKirbylover** [comment]
> All in all, this is more doable than it looks and I could look back into finishing it, although I wouldn't bother if there was no interest in getting RetroAchievements into the upstream - the RA community has mostly abandoned the idea of "RetroAchievements forks" for emulators, which is a big reason why I had decided the first step for returning to this should've been getting a "go-ahead" for submitting a PR.

please please please please oh Please. i would be enthusiastic. i dont usually care for achievements but i think they provide a good incentive to replay a game from start to finish again especially after so long. oh pleasies

**Comment by ZAIDORZ** [comment]
As someone who doesn't use RA as much as other people do, I love the concept and execution of game achievements. Even though if I probably wouldn't make use of an integration like it, it's evident that there are many other people who would love to see this integration come to fruition. I understand that MelonDS has been getting more and more ambitious in terms of features, but I really do think that a feature like this would be a welcome addition to everyone using the emulator.

**Comment by NitroShellMKDS** [comment]
You can even use this library to speed up development of adding this feature, it's C++ and C just like melonDS
https://github.com/RetroAchievements/RALibretro

**Comment by Arisotura** [comment]
this appears to be popular request, so I guess I'll look into it.

my concerns are mainly, from a technical standpoint, how much complexity this would add to the emulator core. if it's reasonable, sure, why not.

I previously said "what's the point", but I guess that's a bias - I personally don't care much about achievements in general. but, well, we're all different, etc.

can't give an ETA on this. like most things, it's mostly "whenever my brain wants to"...

**Comment by Amnesia1000** [comment]
> this appears to be popular request, so I guess I'll look into it.
> 
> my concerns are mainly, from a technical standpoint, how much complexity this would add to the emulator core. if it's reasonable, sure, why not.
> 
> I previously said "what's the point", but I guess that's a bias - I personally don't care much about achievements in general. but, well, we're all different, etc.
> 
> can't give an ETA on this. like most things, it's mostly "whenever my brain wants to"...

Thank you, we are very grateful even if it's just for trying.

**Comment by ksfaltus** [comment]
> this appears to be popular request, so I guess I'll look into it.
> 
> my concerns are mainly, from a technical standpoint, how much complexity this would add to the emulator core. if it's reasonable, sure, why not.
> 
> I previously said "what's the point", but I guess that's a bias - I personally don't care much about achievements in general. but, well, we're all different, etc.
> 
> can't give an ETA on this. like most things, it's mostly "whenever my brain wants to"...

FWIW the MelonDS core on Retroarch does support Retroachievements; it may be worth contacting whoever maintains it

**Comment by JesseTG** [comment]
> FWIW the MelonDS core on Retroarch does support Retroachievements; it may be worth contacting whoever maintains it

No need, I'm active in the Discord already. The caveat is that the hardest parts are done by RetroArch itself; all I need to do is expose some memory. Library integrations, image rendering, service access...that's not handled by the core.

**Comment by PanMenel** [comment]
Well here is a late gift for christmas yall #2529 it is not proffesional but I think it should work for basic needs :D

**Comment by DarkLord4563** [comment]
plus for support

**Comment by SlyAceZeta** [comment]
Playing DS games with RetroAchievements is literally the only reason I have RetroArch at all, since I can't stand its interface. I much prefer standalone emulators. melonDS not having RetroAchievements support is pretty much the only reason I'm not using melonDS; adding it would make me switch in a heartbeat.

**Comment by Optekah** [comment]
Every time I get the email there's activity for this issue, my heart breaks just a little knowing it's probably not a notification about a successful implementation. The open PR has conflicts that must be resolved before merging can happen; is anyone smarter than me able to say if they are resolvable or nah?

**Comment by SlyAceZeta** [comment]
> The open PR has conflicts that must be resolved before merging can happen; is anyone smarter than me able to say if they are resolvable or nah?

Anything is resolvable, but the biggest problem with that PR is that its author used AI to code the entire thing. The fact that they claimed to be "proud" of it makes me laugh, tbh. It probably needs to be re-coded from scratch to get rid of the AI use.

**Comment by NitroShellMKDS** [comment]
> Every time I get the email there's activity for this issue, my heart breaks just a little knowing it's probably not a notification about a successful implementation. The open PR has conflicts that must be resolved before merging can happen; is anyone smarter than me able to say if they are resolvable or nah?

FR

**Comment by ocadigo** [comment]
We NEED RetroAchievements for MelonDS!

**Comment by KprojectsKurt** [comment]
I just hope the project isn't dead

**Comment by PanMenel** [comment]
> > The open PR has conflicts that must be resolved before merging can happen; is anyone smarter than me able to say if they are resolvable or nah?
> 
> Anything is resolvable, but the biggest problem with that PR is that its author used AI to code the entire thing. The fact that they claimed to be "proud" of it makes me laugh, tbh. It probably needs to be re-coded from scratch to get rid of the AI use.

Bruh, are we really still crying about ai code, yeah it makes a problem for developers because well AI sometimes writes code that is too different from source and is stupid to merge. As for players there is no difference, i could just not say that I used it and you would be so happy damn. I can fix some stuff and revise the code so it will resemble more of a commit than a concept but shit like this makes me wonder if yall even deserve that. Ai code is not some magic that is entirely diffrent from human coding it just makes mistakes and stupid decisions, revised ai code when ai uses libraries and wikis of source material, after human fixes is as reliable as fully human, in short in my case I am new to coding and to the scene, so my code will not be on standard but I try, and yes I am proud because it still required a lot of work, it is not some magic prompt that makes stuff. Make it yourself if you think its so easy.

**Comment by PanMenel** [comment]
Also project was initially made to test if I am capable of doing something so many people wanted. I just did it as a challenge, I can update it but because I am not unemployed anymore it can take some more time.

**Comment by PanMenel** [comment]
P.S you can join my discord, there is a channel there to discuss my projects if you want to help out or give feedback.

**Comment by NitroShellMKDS** [comment]
> > > The open PR has conflicts that must be resolved before merging can happen; is anyone smarter than me able to say if they are resolvable or nah?
> > 
> > 
> > Anything is resolvable, but the biggest problem with that PR is that its author used AI to code the entire thing. The fact that they claimed to be "proud" of it makes me laugh, tbh. It probably needs to be re-coded from scratch to get rid of the AI use.
> 
> Bruh, are we really still crying about ai code, yeah it makes a problem for developers because well AI sometimes writes code that is too different from source and is stupid to merge. As for players there is no difference, i could just not say that I used it and you would be so happy damn. I can fix some stuff and revise the code so it will resemble more of a commit than a concept but shit like this makes me wonder if yall even deserve that. Ai code is not some magic that is entirely diffrent from human coding it just makes mistakes and stupid decisions, revised ai code when ai uses libraries and wikis of source material, after human fixes is as reliable as fully human, in short in my case I am new to coding and to the scene, so my code will not be on standard but I try, and yes I am proud because it still required a lot of work, it is not some magic prompt that makes stuff. Make it yourself if you think its so easy.

I hate when people hide Ai coding, like the Mario Kart Wii recomp. You did the right thing PanMenel for always being open about it, that deserves respect. I've been using your fork since you first published it, sure it has problems where it freezes for a second, but I will gladly use it just for RetroAchievements. If AI gets you into learning how to code then so be it.

**Comment by SlyAceZeta** [comment]
Let me just quote @Arisotura here, since she runs melonDS:

> I'm not a fan at all of the fact this is made with AI. I don't really trust AI to produce working or clean code, and I think the time spent to clean up the mess it can make might as well be spent coding this from the ground up without AI.
> 
> While I appreciate the intent, I'm not willing to merge AI code into melonDS.

https://github.com/melonDS-emu/melonDS/pull/2529#issuecomment-3772758585

> To meet my expectations, it'd have to shed all the AI code.

https://github.com/melonDS-emu/melonDS/pull/2529#issuecomment-3773969192

> I feel bad about having so many issues, pull requests, etc, staying open and rotting in the backlog. At the same time, I just don't have the brain power to review every single PR alone.
> 
> And to make matters worse, now we have people submitting AI-generated code, which is problematic from several standpoints: code quality (AI tends to make a mess and not really understand what it's doing, and someone who's using AI isn't likely to understand those things any better), plagiarism, the usual ethical and environmental problems with AI too...

https://melonds.kuribo64.net/comments.php?id=261

tl;dr That AI-generated PR is not getting added.

It's absolutely astounding that you would claim you made that PR "to test if I am capable of doing something so many people wanted" because using AI proves that, actually, you are NOT capable of doing that. To then advertise your Discord server is hilarious and I wouldn't be surprised if @Arisotura just ends up rejecting that PR entirely. We're watching a one-man clown show and every instance just makes me laugh harder. 🍿

> Make it yourself if you think its so easy.

Honestly, this entire thing does have me motivated to write a new RetroAchievements integration into melonDS from scratch without any AI assistance, if no one else beats me to that punch.

The only problem is, in that last link I provided, @Arisotura is considering reorganizing the whole system here so it's not solely on her shoulders, and there's 130 other open PRs still waiting in the queue as well. It might be a bad idea to introduce a new PR for RetroAchievements with such a large queue and before a potential reorganization completely changes the dev process for melonDS. I can stick with RetroArch for now, but DS RetroAchievements really is the only reason I use RetroArch at all. I really do look forward to the day when we can use melonDS for RA, and without that being implemented through plagiarism, environmental damage, and so-called "developers" who don't even know what they're writing and have the gall to advertise themselves.

**Comment by NitroShellMKDS** [comment]
> Honestly, this entire thing does have me motivated to write a new RetroAchievements integration into melonDS from scratch without any AI assistance, if no one else beats me to that punch.

Please do

> I really do look forward to the day when we can use melonDS for RA, and without that being implemented through plagiarism, environmental damage.

they never said they weren't hosting the AI themselves.
Not against what Arisotura wants, it's their project. just saying that PanMenel never stated that they were using AI not hosted by themselves, but I'd like to know since they are already so open about using AI

**Comment by NitroShellMKDS** [comment]
> Let me just quote [@Arisotura](https://github.com/Arisotura) here, since he runs melonDS:

Not to be mean but @Arisotura uses She/Her pronouns, they are a transfem. If you do have the respect for @Arisotura you would use their preferred pronouns.

**Comment by SlyAceZeta** [comment]
> Not to be mean but [@Arisotura](https://github.com/Arisotura) uses She/Her pronouns. they are trans.

AHHH I'm so sorry!! I really should have checked first. Fixed my reply. I'm so sorry for the mistake.

**Comment by NitroShellMKDS** [comment]
> > Not to be mean but [@Arisotura](https://github.com/Arisotura) uses She/Her pronouns. they are trans.
> 
> AHHH I'm so sorry!! I really should have checked first. Fixed my reply. I'm so sorry for the mistake.

Don't say sorry to me, say sorry to @Arisotura

**Comment by SlyAceZeta** [comment]
> > > Not to be mean but [@Arisotura](https://github.com/Arisotura) uses She/Her pronouns. they are trans.
> > 
> > 
> > AHHH I'm so sorry!! I really should have checked first. Fixed my reply. I'm so sorry for the mistake.
> 
> Don't say sorry to me, say sorry to [@Arisotura](https://github.com/Arisotura)

That was what I was trying to do, I'm sorry Arisotura

**Comment by NitroShellMKDS** [comment]
> Not

This has gotten out of hand, this is way offtopic from the main github issue. My fault.
